### PR TITLE
Deployment guide: restructured to add section on HTTPS/SSL configuration | Remove TAR/ZIP deployment option | Upgrading from 2.x guide: Recommend merging src/java into src/groovy

### PR DESCRIPTION
**Pull request by mjcmatrix**
Enhanced deployment guide into sections (standalone, container deployment and Deployment Configuration Tasks).

Added a new section on HTTPS and SSL certificate configuration for the embedded container.

Clarified that `grails run-app -https` is intended only for dev use and generates a self-signed cert (now that `grails prod run-app` is offered as a deployment method).

### CODE REVIEW

File: src/en/guide/deployment/deploymentStandalone.gdoc [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+You can then run either the WAR file or the JAR using your Java installation:
+
+{code}
+java -Dgrails.env=prod -jar build/libs/mywar-0.1.war    (or .jar)
+{code}
+
+h3. A TAR/ZIP distribution
+
+The [package|commandLine] will also produce a TAR and a ZIP file in the @build/distributions@ directory. If you extract these archives (typically the TAR on Unix systems and the ZIP on Windows) you can then run bash file which is the name of your application located in the @bin@ directory.
```
Comment (graemerocher) [inline src/en/guide/deployment/deploymentStandalone.gdoc]: This is not true in 3.1 and it not really a supported feature, it was an added side effect of using the Gradle `application` plugin which is no longer used in 3.1

**Comment by graemerocher** [comment]
Please send this pull request to the 3.0.x branch. Thanks.

File: src/en/guide/deployment/deploymentStandalone.gdoc (hunk 1 above)
Comment (mjcmatrix) [inline src/en/guide/deployment/deploymentStandalone.gdoc]: Hi @graemerocher, what is not a supported feature? The Spring Boot SSL settings, or something else?  I added the SSL section; everything else is just existing master content I have restructured.

Comment (graemerocher) [inline src/en/guide/deployment/deploymentStandalone.gdoc]: The creation of a TAR/ZIP distribution. It is a feature of the Gradle `application` plugin, which happened to be used by 3.0.x, but not by 3.1.x

Comment (mjcmatrix) [inline src/en/guide/deployment/deploymentStandalone.gdoc]: Yes I didn't add this doco, it was already in master. Shall I correct this in the PR and submit another?

Comment (graemerocher) [inline src/en/guide/deployment/deploymentStandalone.gdoc]: Probably a good idea, thanks for the contributions!

**Comment by mjcmatrix** [comment]
Ready for merge. Removed TAR/ZIP distribution standalone deployment option from master.

**Comment by mjcmatrix** [comment]
@graemerocher will this still be committed to master as amended?

**Comment by mjcmatrix** [comment]
This PR now contains an improvement to the Upgrading from 2.x guide: Recommend merging src/java into src/groovy and make clear this not a typo in the docs (based on developer feedback); plus a few more minor improvements to the two updated sections to make added content clearer and more concise.

**Comment by mjcmatrix** [comment]
@graeme: Can you please consider this pull request for **master**.

It contains three changes:
    1. Deployment guide: Restructured to add new section on HTTPS/SSL configuration
    2. Remove TAR/ZIP deployment option (which was still in master)
    3. In Upgrading from 2.x guide: Recommend merging src/java into src/groovy during migration

A separate PR was made for the grails-doc 3.0.x branch containing only Item 1 and already accepted.

**Pull request merged**